routes/category.js

```diff
@@ -1,26 +1,25 @@
 const express = require('express');
 const router = express.Router();
-const products = require('../src/data/products');
-const categories = require('../src/data/categories');
+const service = require('../services/dataService');
 
+/*
+ * This method first gets the category ID from the request params.
+ * Then it loops through the combinedProductMap.
+ * It checks if the category id of that particular product is equal to the request param
+ * If matches then push to the response array.
+ */
 router.get('/:ctyId', function(req, res, next) {
-    const ctyId = req.params.ctyId;
-    let currentCategory;
-    let response = [];
-    categories.categories.forEach( category => {
-        if (category.id === ctyId) {
-            currentCategory = category;
-        }
-    });
-    products.products.forEach( product => {
-        if (product.categoryId === ctyId) {
-            response.push({
-                ...product,
-                category: currentCategory
-            });
-        }
-    });
-    res.send(response);
+  const { ctyId } = req.params;
+  const response = [];
+  const combinedProductMap = service.getCombinedProductMap();
+  for (const product in combinedProductMap) {
+    if (combinedProductMap.hasOwnProperty(product)) {
+      if (combinedProductMap[product].categoryId === ctyId) {
+        response.push(combinedProductMap[product]);
+      }
+    }
+  }
+  res.send(response);
```

Comment:
The response should be a map like [your `products/all`](https://storage.googleapis.com/codein-prod.appspot.com/gci-2019/core_taskupdate/doc/4819050333470720_1578203679_Screenshot_2020-01-05_at_9.48.19_AM.png?Expires=1578391630&GoogleAccessId=codein-prod%40appspot.gserviceaccount.com&Signature=eu5JrgYk8ezC3chdmZ3OWHCuz7PXCGx6AOgh9KfWLW3%2FJFUZul6dX%2BHq%2F9W54mowEIx%2BpOYRvwKmOyR17FVPXN0M2Y1oGc4jqOOHFT2Fvgc4mI05tI0AAd56wiaW5JJ%2BR7B5r53sQ%2BRhZnvkP7%2FLfikA8FQheYHCTkgiIuAoOauZfVTZ1CjBFQNAUkGK5BxiQDTmvqA72puTs6uJMPVum0bSm7hR4SX6GOHdaaRL44zj%2Fa9E23quYKzxuarP2ACT39w%2BgVc8Uz6eko18HERiT9gMdPI5NbYJRud49oywOxsErdoTxzDKBRKOPQCcIpdODXdA45ykubHtJUJ4vLkbZA%3D%3D) response, just filtered to match the input category id.